### ISSUE TITLE
matRad caught in endless loop in setCstTable when reopening

### ISSUE THREAD

**Issue by wahln** [e0404/matRad#139]
When I reopen the GUI with the current dev branch commit, matRad seems to be caught in an endless loop sometimes. When I terminate, this is the location.

```
Operation terminated by user during matRadGUI>setCstTable (line 1630)

In matRadGUI_matRadGUI_OpeningFcn (line 350)
        cst = setCstTable(handles,cst);

In gui_mainfcn (line 220)
    feval(gui_State.gui_OpeningFcn, gui_hFigure, [], guidata(gui_hFigure), varargin{:});

In matRadGUI (line 71)
    gui_mainfcn(gui_State, varargin{:});
```

This seems to be the issue:
```
colorAssigned = true;

while colorAssigned == true
  for i = 1:size(cst,1)
    if ~isfield(cst{i,5},'visibleColor')
      colorAssigned = false;
    elseif isempty(cst{i,5}.visibleColor)
      colorAssigned = false;
    end
  end
end
```

Seems that calling setCstTable when all colors are already set will get stuck in this loop? I do not quite understand this part of the code.

**Comment by markbangert** [comment]
How can we try to reproduce this behaviour?

**Comment by markbangert** [comment]
@Haydnspass shouldn't these lines read

for i = 1:size(cst,1)
    if ~isfield(cst{i,5},'visibleColor')
      colorAssigned = false;
      break;
    elseif isempty(cst{i,5}.visibleColor)
      colorAssigned = false;
      break;
    end
end

**Comment by AlexS1992** [comment]
I think the case where it does not terminate is the one where all structures **have** a assigned color, because then the loop will never hit any of the colorAssigned = false statements.

What is actually the purpose of the [while loop](https://github.com/e0404/matRad/blob/dev/matRadGUI.m#L1628) ?

Without the while loop I think this code would run fine, because we would check every element of the cst for a color and if one is missing we use the color cube instead.
Am I missing something here?

**Comment by Haydnspass** [comment]
@markbangert yep, that's exactly what was in my mind, but I was not aware of the break statement at this time. Opened a pull request.

**Comment by markbangert** [comment]
thank you all for taking care of this so quickly! just merged pull request #141 .

will close this issue now!
